frontend/pages/admin/IntegrationsPage/cards/ConditionalAccess/ConditionalAccess.tsx

```diff
@@ -242,173 +227,210 @@ const ConditionalAccess = () => {
+    // Don't check config if we're showing the awaiting OAuth banner
+    if (showAwaitingOAuthBanner) {
+      setEntraPhase(EntraPhase.NotConfigured);
+      return;
+    }
+
+    if (entraTenantId) {
+      if (!entraConfigured) {
+        setEntraPhase(EntraPhase.ConfirmingConfigured);
       } else {
         // tenant id is present and connection is configured
-        setPhase(Phase.Configured);
+        setEntraPhase(EntraPhase.Configured);
       }
+    } else {
+      setEntraPhase(EntraPhase.NotConfigured);
     }
-  }, [contextConfigMsetId, contextConfigMseConfigured]);
+  }, [entraTenantId, entraConfigured, showAwaitingOAuthBanner]);
 
   if (!isPremiumTier) {
     return <PremiumFeatureMessage />;
   }
 
   // HANDLERS
 
-  const toggleDeleteConditionalAccessModal = () => {
-    setShowDeleteConditionalAccessModal(!showDeleteConditionalAccessModal);
+  const toggleDeleteModal = () => {
+    setShowDeleteModal(!showDeleteModal);
   };
 
-  const onSubmit = async (evt: React.FormEvent<HTMLFormElement>) => {
-    evt.preventDefault();
+  const handleEntraConnect = () => {
+    setShowEntraModal(true);
+  };
 
-    const errs = validate(formData);
-    if (Object.keys(errs).length > 0) {
-      setFormErrors(errs);
-      return;
-    }
-    setIsUpdating(true);
-    try {
-      const {
-        microsoft_authentication_url: msAuthURL,
-      } = await conditionalAccessAPI.triggerMicrosoftConditionalAccess(
-        formData[MSETID]
-      );
-      setIsUpdating(false);
-      setPhase(Phase.FormSubmitted);
-      window.open(msAuthURL);
-    } catch (e) {
-      renderFlash(
-        "error",
-        "Could not update conditional access integration settings."
-      );
-      setIsUpdating(false);
-    }
+  const handleEntraModalClose = () => {
+    setShowEntraModal(false);
   };
 
-  const onDeleteConditionalAccess = async () => {
-    setFormData({ [MSETID]: "" });
-    refetchConfig();
+  const handleEntraModalSuccess = () => {
+    setShowEntraModal(false);
+    // Show banner instead of immediately refetching config
+    // Config will be checked when user refreshes the page
+    setShowAwaitingOAuthBanner(true);
   };
 
-  const onInputChange = ({ name, value }: IInputFieldParseTarget) => {
-    const newFormData = { ...formData, [name]: value };
-    setFormData(newFormData);
-    const newErrs = validate(newFormData);
-    // only set errors that are updates of existing errors
-    // new errors are only set onBlur or submit
-    const errsToSet: Record<string, string> = {};
-    Object.keys(formErrors).forEach((k) => {
-      // @ts-ignore
-      if (newErrs[k]) {
-        // @ts-ignore
-        errsToSet[k] = newErrs[k];
-      }
-    });
-    setFormErrors(errsToSet);
+  const onDeleteConditionalAccess = (updatedConfig: IConfig) => {
+    setConfig(updatedConfig);
+    setIsUpdating(false);
   };
 
-  const onInputBlur = () => {
-    setFormErrors(validate(formData));
+  const handleOktaConnect = () => {
```

Comment:
Same comments about this and below functions as for above modal opening/closing behavior